tests/devices/test_default_qubit_jax.py

```diff
@@ -0,0 +1,462 @@
+        x = 0.43316321
+        y = 0.2162158
+        z = 0.75110998
+        weights = jnp.array([x, y, z])
+
+        dev = qml.device("default.qubit.jax", wires=1)
+
+        @qml.qnode(dev, interface="jax")
+        def circuit(p):
+            qml.RX(3 * p[0], wires=0)
+            qml.RY(p[1], wires=0)
+            qml.RX(p[2] / 2, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        if not qml.tape_mode_active():
+            assert isinstance(circuit, qml.qnodes.PassthruQNode)
+        res = circuit(weights)
+
+        expected = jnp.cos(3 * x) * jnp.cos(y) * jnp.cos(z / 2) - jnp.sin(3 * x) * jnp.sin(z / 2)
+        assert jnp.allclose(res, expected, atol=tol, rtol=0)
+
+        grad_fn = jacobian_transform(circuit, 0)
+        res = grad_fn(jnp.array(weights))
+
+        expected = jnp.array(
+            [
+                -3
+                * (jnp.sin(3 * x) * jnp.cos(y) * jnp.cos(z / 2) + jnp.cos(3 * x) * jnp.sin(z / 2)),
+                -jnp.cos(3 * x) * jnp.sin(y) * jnp.cos(z / 2),
+                -0.5
+                * (jnp.sin(3 * x) * jnp.cos(z / 2) + jnp.cos(3 * x) * jnp.cos(y) * jnp.sin(z / 2)),
+            ]
+        )
+
+        assert jnp.allclose(res, expected, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("jacobian_transform", [jax.jacfwd, jax.jacrev])
+    def test_jacobian_repeated(self, tol, jacobian_transform):
+        """Test that jacobian of a QNode with an attached default.qubit.jax device
+        gives the correct result in the case of repeated parameters"""
+        x = 0.43316321
+        y = 0.2162158
+        z = 0.75110998
+        p = jnp.array([x, y, z])
+        dev = qml.device("default.qubit.jax", wires=1)
+
+        @qml.qnode(dev, interface="jax")
+        def circuit(x):
+            qml.RX(x[1], wires=0)
+            qml.Rot(x[0], x[1], x[2], wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        res = circuit(p)
+
+        expected = jnp.cos(y) ** 2 - jnp.sin(x) * jnp.sin(y) ** 2
+        assert jnp.allclose(res, expected, atol=tol, rtol=0)
+
+        grad_fn = jacobian_transform(circuit, 0)
+        res = grad_fn(p)
+
+        expected = jnp.array(
+            [-jnp.cos(x) * jnp.sin(y) ** 2, -2 * (jnp.sin(x) + 1) * jnp.sin(y) * jnp.cos(y), 0]
+        )
+        assert jnp.allclose(res, expected, atol=tol, rtol=0)
+
+    def test_state_differentiability(self, tol):
+        """Test that the device state can be differentiated"""
+        dev = qml.device("default.qubit.jax", wires=1)
+
+        @qml.qnode(dev, diff_method="backprop", interface="jax")
+        def circuit(a):
+            qml.RY(a, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        a = jnp.array(0.54)
+
+        def cost(a):
+            """A function of the device quantum state, as a function
+            of ijnput QNode parameters."""
+            circuit(a)
+            res = jnp.abs(dev.state) ** 2

```

Comment:
We've been migrating `state` to be a valid QNode return class, rather than this previous device attribute usage, but nice to know that it works!

Reply:
I don't think we need to remove this test, but it would be good to have an equivalent one for `return qml.state`

Reply:
Will do. 

I think we should also remove the `dev.state` attribute. It's not really functional.

Reply:
Test added.

Reply:
:100: